python-avd/pyavd/_eos_designs/schema/schema_fragments/defs_network_services.schema.yml

```
@@ -508,7 +508,13 @@ $defs:
                     description: If not set, "vrf_id" will be used.
                   router_id:
                     type: str
-                    description: If not set, switch router_id will be used.
+                    description: |-
+                      Router ID to use for OSPF in this VRF.
+                      This can be an IPv4 address, "main_router_id", "none" or "vtep_diagnostic".
+                      - "main_router_id" will reuse the main OSPF Router ID which is the IP address of Loopback0.
+                      - "none" will not configure a OSPF Router ID for this VRF. EOS will use the main OSPF Router ID.
+                      - "vtep_diagnostic" will use the IP address of the VRF Diagnostic Loopback interface.
```

I would like to get rid of the `vtep_` part of the vtep_diagnostic stuff, since it also applies to non-VTEP scenarios. So let's call this something else:

```suggestion
                      - "diagnostic_loopback" will use the IP address of the VRF Diagnostic Loopback interface.
```

---

same for bgp

---

Updated at both places. Updated the logic in utils.py as well from `if router_id == "vtep_diagnostic"` to `if router_id == "diagnostic_loopback"`.